src/main/kotlin/org/jitsi/nlj/util/StreamInformation.kt

```diff
@@ -67,4 +81,25 @@ class StreamInformationStoreImpl : StreamInformationStore {
             _rtpExtensions.find { it.type == rtpExtensionType }?.let { handler(it.id.toInt()) }
         }
     }
+
+    override fun addRtpPayloadType(payloadType: PayloadType) {
+        synchronized(payloadTypesLock) {
+            _rtpPayloadTypes[payloadType.pt] = payloadType
+            payloadTypeHandlers.forEach { it(_rtpPayloadTypes) }
```

Comment:
Should we call the handlers outside the synchronized block?

Reply:
i looked into this and think doing that could result in a 'double event'.  for  example, calls to addRtpExtensionMapping and onRtpExtensionMapping which happened in the right order could cause a double event: addRtpExtensionMapping adds a new mapping, releases lock, pre-empted before invoking handlers, onRtpExtensionMapping adds the handler, releases the lock, notifies the new handler, addRtpExtensionMapping scheduled again, notifies all handlers (including the new one who already got the updated state
with the current code, at least, i don't think this would result in a bug. but maybe some confusing prints or something?